client/state/document/reducer.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SET_DOCUMENT_DESCRIPTION,
+	SET_DOCUMENT_TITLE,
+} from 'state/action-types';
+
+const INITIAL_STATE = {
+	title: '',
+	description: '',
+};
+
+export function title( state = '', action ) {
+	switch ( action.type ) {
+		case SET_DOCUMENT_TITLE:
+			return action.title;
+	}
+
+	return state;
+}
+
+export function description( state = '', action ) {
```

Comment:
I think we can keep the `setDescription` action creator and `DOCUMENT_DESCRIPTION_SET` action type, but I think we don't need a separate reducer. Instead, implement the reducer of `meta` such that we capture the `DOCUMENT_DESCRIPTION_SET` action type and transform it to its equivalent meta properties.